### PR TITLE
Dirty Algo Frontend PR

### PR DESCRIPTION
**Pull request by barkincavdaroglu**
What changed?
Diffusion centrality used to return the results for last time step. Now it returns results for all of the time steps (i.e. if t=4, then we return diffusion centrality PER node PER timestep).

AnalyticsPane.tsx is copied/pasted from radixui but with different content ([link](https://www.radix-ui.com/docs/primitives/components/accordion) but select tailwind).

Each of the algorithm views/components expect redux data coming through. These are represented as:

1. Algorithm result,
2. whether the algorithm is set to "Running" or "Not Running" (boolean),
3. Function to change whether to run or not run the algorithm (function),

Diffusion centrality component is dirty because I couldn't find a way to pass 3D hashmap in TypeScript (gave errors unless I put the data inside the function, discussion for later).

The type of data each algorithm expects/returns is explicitly stated.

Images:
Screenshot 2023-03-03 at 19 54 56

Screenshot 2023-03-03 at 19 55 13

Screenshot 2023-03-03 at 19 55 19

LOCATION is subject to change.

### CODE REVIEW

File: src/components/Map/algorithms/Chart.tsx
Comment (barkincavdaroglu) [inline src/components/Map/algorithms/Chart.tsx:41]: This is what I expect would come from redux store. But if you isolate:
```js
const diffcen = {...}
```
and pass it to convertToDataFormat, you will get errors saying the type doesn't have certain types of functionalities.

File: src/components/Map/algorithms/Chart.tsx
Comment (barkincavdaroglu) [inline src/components/Map/algorithms/Chart.tsx:68]: Object.entries(dictionary) is the only way this works that I could come up with.

File: src/components/Map/algorithms/Chart.tsx
Comment (barkincavdaroglu) [inline src/components/Map/algorithms/Chart.tsx:105]: Nothing too important, its just that the color of the node with largest diffusion centrality supersedes the rest of nodes if we don't reverse the dataset. This is purely for cosmetics.

File: src/components/Map/algorithms/Chart.tsx
Comment (barkincavdaroglu) [inline src/components/Map/algorithms/Chart.tsx:114]: Code taken from: https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb

**Comment by barkincavdaroglu** [comment]
The errors are due to some stuff in radixui as I copy-pasted their code for Accordion. These errors are not breaking errors but "warnings".

File: src-tauri/src/analytics/algo_result_enums/diff_cen.rs
Comment (uhuruhashimoto) [inline src-tauri/src/analytics/algo_result_enums/diff_cen.rs:5]: Why has this changed from a hashmap of hashmaps to a hashmap of hashmap of hashmaps? ...This is the best sentence I've ever typed in my life.

Comment (uhuruhashimoto) [reply src-tauri/src/analytics/algo_result_enums/diff_cen.rs:5]: If I'm understanding correctly, each timestep maps to a full set of nodes (that's our first hashmap). Each of those nodes maps to a bunch of floats that's a diffcen result (that's our second hashmap).

Comment (uhuruhashimoto) [reply src-tauri/src/analytics/algo_result_enums/diff_cen.rs:5]: Anyways you'll get serialization when you pull from my PR (merged), which may fix some of the errors.

Comment (barkincavdaroglu) [reply src-tauri/src/analytics/algo_result_enums/diff_cen.rs:5]: Yeah we are returning all timestep results now for the chart so that user can see how diffusion changes over time

File: src-tauri/src/analytics/algo_result_enums/diff_cen.rs
Comment (uhuruhashimoto) [inline src-tauri/src/analytics/algo_result_enums/diff_cen.rs:3]: This result is serialized on the main branch as of now (`#[derive(Debug, Serialize, Deserialize)]`), which might help with typescript errors. I don't think we can pass the data from the backend to the front via command without serializing.

Comment (barkincavdaroglu) [reply src-tauri/src/analytics/algo_result_enums/diff_cen.rs:3]: Yeah I will pull from the main soon.

Comment (barkincavdaroglu) [reply src-tauri/src/analytics/algo_result_enums/diff_cen.rs:3]: Typescript errors come from the accordion elements. IDK what's going on with those

**Comment by uhuruhashimoto** [comment]
Mind adding your package.json file to the PR? I'm getting this error when I pull and run the branch
Screen Shot 2023-03-06 at 10 03 04 PM

File: src/components/Map/AnalyticsPane.tsx
Comment (uhuruhashimoto) [inline src/components/Map/AnalyticsPane.tsx:17]: what do you mean by dummy function?

File: src/components/Map/MapView.tsx
Comment (uhuruhashimoto) [inline src/components/Map/MapView.tsx:111]: Have we made an overflow behavior decision about the accordion vs info panes (whatever's already on the left)?

File: src/components/Map/algorithms/Chart.tsx
Comment (uhuruhashimoto) [reply src/components/Map/algorithms/Chart.tsx:41]: what errors specifically?

File: src/components/Map/algorithms/Chart.tsx
Comment (uhuruhashimoto) [inline src/components/Map/algorithms/Chart.tsx:117]: I'm not super keen on regex without explanation because it's so obfuscated. That said, [here's](https://stackoverflow.com/questions/36686461/why-does-regex-not-match-when-there-are-more-then-one) an explanation of the same GitHub code and a further simplification.

File: package.json
Comment (uhuruhashimoto) [inline package.json:21]: Does this include classnames? I'm getting this error when I pull and run the branch, even after `pnpm i`: 
Screen Shot 2023-03-06 at 10 03 04 PM

Comment (uhuruhashimoto) [reply package.json:21]: updated `package.json` and `pnpm-lock.yaml`

**Review by uhuruhashimoto — COMMENTED** [review]
Here are the things that I think we need for this PR to go in:

General thing:
We need to make a design decision about whether to display live algos results. If so, we'll need some testing. I'm getting an empty result for all algos right now, and we'd need to check whether we'll get one live. Our graph weights are probably off and will need debugging. That's untested at this point.

Technical things:
- A "run all algos" button that passes the booleans to the backend (for bitfield)
- Hook up the accordion to a button (layers button maybe)
- overflow behavior/priority defined if we click on a node and have network behavior open as well

**Comment by sherry1iu** [comment]
image
We have a conflict with ordering on screen -

**Comment by barkincavdaroglu** [comment]
> image We have a conflict with ordering on screen -

Yeah the current place was a placeholder. I was going to use the layer button on top right to open/close the network analytics pane.

**Comment by barkincavdaroglu** [comment]
I have a final presentation that I have been preparing for but I will do my best to complete the implementation right after (and right before technigala). @uhuruhashimoto @sherry1iu

**Comment by sherry1iu** [comment]
I can do the toggling if you want, it's pretty similar to what I did for waypoints. @barkincavdaroglu

**Comment by sherry1iu** [comment]
image
Other thing is to make sure to update all the accordion text

**Comment by uhuruhashimoto** [comment]
> image Other thing is to make sure to update all the accordion text

Done

**Comment by barkincavdaroglu** [comment]
> I can do the toggling if you want, it's pretty similar to what I did for waypoints. Lmk @barkincavdaroglu

Can you please? Thank you so much 🥺

**Comment by sherry1iu** [comment]
accordion

Would like a second pair of eyes

**Comment by barkincavdaroglu** [comment]
> accordion [ accordion ](https://user-images.githubusercontent.com/57198797/223451554-70e19489-2933-4cfc-b182-d91e47c3eb78.gif) [ ](https://user-images.githubusercontent.com/57198797/223451554-70e19489-2933-4cfc-b182-d91e47c3eb78.gif)
> 
> Would like a second pair of eyes

Looks amazing!!

**Comment by ajmcquilkin** [comment]
Just pushed a styling and refactoring commit to remove the linting errors, should unblock the CI for merge once we're ready

**Comment by sherry1iu** [comment]
image
button

**Comment by sherry1iu** [comment]
i have sausage fingers my bad

File: src/components/Map/MapView.tsx
Comment (uhuruhashimoto) [reply src/components/Map/MapView.tsx]: resolved: last clicked on is displayed.

File: src/components/Map/AnalyticsPane.tsx
Comment (uhuruhashimoto) [reply src/components/Map/AnalyticsPane.tsx]: resolved: sherry added a run button.

**Review by uhuruhashimoto — APPROVED** [review]
let's get this money

**Pull request merged**
